src/plugins/teleop/Teleop_TEST.cc

```diff
@@ -0,0 +1,354 @@
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif
+#include <ignition/msgs/stringmsg.pb.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "ignition/gui/Application.hh"
+#include "ignition/gui/Plugin.hh"
+#include "ignition/gui/MainWindow.hh"
+#include "ignition/gui/qt.h"
+
+#include "Teleop.hh"
+
+int g_argc = 1;
+char **g_argv = new char *[g_argc];
+
+using namespace ignition;
+using namespace gui;
+
+class TeleopTest : public ::testing::Test
+{
+  // Provides an API to load plugins and configuration files.
+  protected: Application app{g_argc, g_argv};
+
+  // List of plugins.
+  protected: QList<plugins::Teleop *> plugins;
+  protected: plugins::Teleop * plugin;
+
+  // Instance of the main window.
+  protected: MainWindow * win;
+
+  // Checks if a new command has been received.
+  protected: bool received = false;
+  protected: transport::Node node;
+
+  // Define velocity values.
+  protected: const double linearVel = 1.0;
+  protected: const double angularVel = 0.5;
+
+  // Set up function.
+  protected:
+    void SetUp() override
+    {
+      common::Console::SetVerbosity(4);
+
+      this->app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+      // Load plugin
+      const char *pluginStr =
+        "<plugin filename=\"Teleop\">"
+          "<ignition-gui>"
+            "<title>Teleop!</title>"
+          "</ignition-gui>"
+        "</plugin>";
+
+      tinyxml2::XMLDocument pluginDoc;
+      EXPECT_EQ(tinyxml2::XML_SUCCESS, pluginDoc.Parse(pluginStr));
+      EXPECT_TRUE(this->app.LoadPlugin("Teleop",
+          pluginDoc.FirstChildElement("plugin")));
+
+      // Get main window
+      win = this->app.findChild<MainWindow *>();
+      ASSERT_NE(nullptr, win);
+
+      // Show, but don't exec, so we don't block
+      win->QuickWindow()->show();
+
+      // Get plugin
+      plugins = win->findChildren<plugins::Teleop *>();
+      EXPECT_EQ(plugins.size(), 1);
+
+      plugin = plugins[0];
+      EXPECT_EQ(plugin->Title(), "Teleop!");
+
+      // Subcribes to the command velocity topic.
+      node.Subscribe("/model/vehicle_blue/cmd_vel",
+          &TeleopTest::CallBack, this);
+
+      // Sets topic. This must be the same as the
+      // one the node is subscribed to.
+      plugin->OnTopicSelection(
+          QString::fromStdString("/model/vehicle_blue/cmd_vel"));
+
+      // Checks if the directions of the movement are set
+      // with the default value '0'.
+      EXPECT_EQ(plugin->LinearDirection(), 0);
+      EXPECT_EQ(plugin->AngularDirection(), 0);
+
+      // Set velocity value and movement direction.
+      plugin->OnLinearVelSelection(QString::number(linearVel));
+      plugin->OnAngularVelSelection(QString::number(angularVel));
+    }
+
+  // Subscriber call back function. Verifies if the Twist message is
+  // sent correctly.
+  protected:
+    void CallBack(const msgs::Twist &_msg)
```

Comment:
nit: `VerifyTwistMsgCb` instead of `CallBack` just to be more verbose about what the method does.